src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php

```diff
@@ -68,15 +68,17 @@ public function process(ContainerBuilder $container)
             if ($pool->isAbstract()) {
                 continue;
             }
+            $class = $adapter->getClass();
             while ($adapter instanceof ChildDefinition) {
                 $adapter = $container->findDefinition($adapter->getParent());
+                $class = $class ?: $adapter->getClass();
                 if ($t = $adapter->getTag($this->cachePoolTag)) {
                     $tags[0] += $t[0];
                 }
             }
             $name = $tags[0]['name'] ?? $id;
             if (!isset($tags[0]['namespace'])) {
-                $tags[0]['namespace'] = $this->getNamespace($seed, $name);
+                $tags[0]['namespace'] = $this->getNamespace($seed, $name.$class);
```

Comment:
I'd suggest a separator here, between the name and the class, just in case.

Reply:
Using which character?

Reply:
a dot :)